packages/server/modules/activitystream/repositories/index.ts

```diff
@@ -270,3 +272,16 @@ export const saveActivityFactory =
       })
     }
   }
+
+export const saveActivityFactory =
+  ({ db }: { db: Knex }): SaveActivity =>
+  async (activity): Promise<{ id: string }> => {
+    const id = cryptoRandomString({ length: 10 })
+
+    await db(Activity.name).insert({
+      ...activity,
+      id
+    })
+
+    return { id }
```

Comment:
use .insert(item, '*') to get the full row back from the DB and return the full entity